tests/test_text.py

```diff
@@ -0,0 +1,464 @@
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        method_unicost = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="unicost",
+                                                   trials=1)
+
+        method_diverse = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="diverse",
+                                                   trials=1)
+
+        selector_unicost = Selective(method_unicost)
+        selector_unicost.fit(data, labels)
+        selected_features_unicost = set(selector_unicost.transform(data).columns)
+
+        selector_diverse = Selective(method_diverse)
+        selector_diverse.fit(data, labels)
+        selected_features_diverse = set(selector_diverse.transform(data).columns)
+
+        assert selected_features_unicost == selected_features_diverse
+
+    def test_text_based_random_unicost(self):
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        method = SelectionMethod.TextBased(num_features=None,
+                                           optimization_method="random",
+                                           cost_metric="unicost",
+                                           trials=5)
+
+        selector = Selective(method)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        # Check whether the selector.transform() is returned a DataFrame or not
+        self.assertTrue(isinstance(selected_features, pd.DataFrame))
+
+        # Check whether the number of row `selected_features' is the same as
+        # the number of rows in the input data or not
+        self.assertEqual(selected_features.shape[0], data.shape[0])
+
+        # Check whether the number of columns in `selected_features' is equal to num_features (user input) or not
+        if method.num_features is None:
+            pass
```

Comment:
so this test is just passing? (num_features is given as None above)

Reply:
Modified all tests for this assert.